src/main/java/hudson/plugins/throttleconcurrents/pipeline/ThrottleStepExecution.java

```diff
@@ -0,0 +1,80 @@
+package hudson.plugins.throttleconcurrents.pipeline;
+
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.plugins.throttleconcurrents.ThrottleJobProperty;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.jenkinsci.plugins.workflow.steps.BodyExecution;
+import org.jenkinsci.plugins.workflow.steps.BodyExecutionCallback;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+
+public class ThrottleStepExecution extends StepExecution {
+    private final ThrottleStep step;
+
+    public ThrottleStepExecution(@Nonnull ThrottleStep step, StepContext context) {
+        super(context);
+        this.step = step;
+    }
+
+    public String getCategory() {
+        return step.getCategory();
+    }
+
+    @Override
+    public boolean start() throws Exception {
```

Comment:
Maybe makes sense to contribute some information to the build log using the BuildListener from the context